### PR TITLE
Fix Choropleth legend with one bucket

### PR DESCRIPTION
**Pull request by ethervoid**
If we receive just one filter we're not showing the ramp color properly
because the `linear_gradient` css property in the template need two
values so we're going to pass the same value and legend twice

### CODE REVIEW

File: src/windshaft-integration/legends/rule-to-choropleth-legend-adapter.js [hunk 1]
```diff
@@ -12,15 +12,23 @@ var isEveryBucketValid = function (rule) {
...
-  return _.map(buckets, function (bucket, i) {
-    var label = '';
-    if (i === 0) {
-      label = bucket.filter.start;
-    } else if (i === buckets.length - 1) {
-      label = bucket.filter.end;
-    }
-    return { value: bucket.value, label: label.toString() };
-  });
+  if (buckets.length === 1) {
```
Comment (IagoLast) [inline src/windshaft-integration/legends/rule-to-choropleth-legend-adapter.js]: could we simplify this a little bit?

```javascript
 function generateColors (buckets) {
  if (buckets.length === 1) {
    var bucket = buckets[0];
    var labelStart = bucket.filter.start;
    var labelEnd = bucket.filter.end;
    return [{ value: bucket.value, label: labelStart.toString() }, { value: bucket.value, label: labelEnd.toString() }];
  }
  return _.map(buckets, function (bucket, i) {
    var label = '';
    if (i === 0) {
      label = bucket.filter.start;
    }
    if (i === buckets.length - 1) {
      label = bucket.filter.end;
    }
    return { value: bucket.value, label: label.toString() };
  });
}
```

Comment (ethervoid) [reply src/windshaft-integration/legends/rule-to-choropleth-legend-adapter.js]: Sure, changing

**Comment by ethervoid** [comment]
@IagoLast made the change, care to review again :)

File: src/windshaft-integration/legends/rule-to-choropleth-legend-adapter.js [hunk 2]
```diff
@@ -12,6 +12,12 @@ var isEveryBucketValid = function (rule) {
 };
 
 var generateColors = function (buckets) {
+  if (buckets.length === 1) {
```
Comment (alonsogarciapablo) [inline src/windshaft-integration/legends/rule-to-choropleth-legend-adapter.js]: Another (crazy?) option here...

```
if (buckets.length === 1) {
  buckets = [ buckets[0], buckets[0] ] 
}
return _.map(buckets, function (bucket, i) {
...

```

Comment (ethervoid) [reply src/windshaft-integration/legends/rule-to-choropleth-legend-adapter.js]: I'd work but IMHO is less readable

**Comment by alonsogarciapablo** [comment]
👏  👏  👏

**Pull request merged**